src/homepageExperience/containers/ArduinoWizard.tsx

```diff
@@ -64,46 +64,38 @@ export class ArduinoWizard extends PureComponent<{}, State> {
   }
 
   handleNextClick = () => {
-    this.setState(
+    this.setState({
+      currentStep: Math.min(
+        this.state.currentStep + 1,
+        HOMEPAGE_NAVIGATION_STEPS_ARDUINO.length
+      ),
+    })
+    event(
+      'firstMile.arduinoWizard.next.clicked',
+      {},
       {
-        currentStep: Math.min(
-          this.state.currentStep + 1,
-          HOMEPAGE_NAVIGATION_STEPS_ARDUINO.length
+        clickedButtonAtStep: normalizeEventName(
+          HOMEPAGE_NAVIGATION_STEPS_ARDUINO[this.state.currentStep - 1].name
+        ),
+        currentStep: normalizeEventName(
+          HOMEPAGE_NAVIGATION_STEPS_ARDUINO[this.state.currentStep].name
         ),
-      },
-      () => {
-        event(
-          'firstMile.arduinoWizard.next.clicked',
-          {},
-          {
-            clickedButtonAtStep: normalizeEventName(
-              HOMEPAGE_NAVIGATION_STEPS_ARDUINO[this.state.currentStep - 1].name
-            ),
-            currentStep: normalizeEventName(
-              HOMEPAGE_NAVIGATION_STEPS_ARDUINO[this.state.currentStep].name
-            ),
-          }
-        )
       }
     )
   }
 
   handlePreviousClick = () => {
-    this.setState(
-      {currentStep: Math.max(this.state.currentStep - 1, 1)},
-      () => {
-        event(
-          'firstMile.arduinoWizard.previous.clicked',
-          {},
-          {
-            clickedButtonAtStep: normalizeEventName(
-              HOMEPAGE_NAVIGATION_STEPS_ARDUINO[this.state.currentStep - 1].name
-            ),
-            currentStep: normalizeEventName(
-              HOMEPAGE_NAVIGATION_STEPS_ARDUINO[this.state.currentStep].name
-            ),
-          }
-        )
+    this.setState({currentStep: Math.max(this.state.currentStep - 1, 1)})
+    event(
```

Comment:
Now that the event function call is out of the callback, its not guranteed what the state of `currentStep` will be. This might get called before/after the state is updated. 

we should move the event call before the setState for that reason. 

Does that make sense? 

https://stackoverflow.com/a/42038724